frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx

```diff
@@ -46,11 +46,35 @@ const ParameterSettings = ({
   onChangeSourceConfig,
   onRemoveParameter,
 }: ParameterSettingsProps): JSX.Element => {
+  const [internalValue, setInternalValue] = useState(parameter.name);
+
+  useLayoutEffect(() => {
+    setInternalValue(parameter.name);
+  }, [parameter.name]);
+
+  const handleChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setInternalValue(event.target.value);
+    },
+    [],
+  );
+
+  const labelInputError = useMemo(() => {
+    if (internalValue === "") {
+      return t`Required`;
+    }
+    return null;
+  }, [internalValue]);
```

Comment:
Done

Reply:
What's a rule of thumb to know when useMemo is reducing performance vs not?

Reply:
> In this case `useMemo` is actually reducing performance due to memoization and hook overhead.

Just to be clear: in this case it's totally negligible, it's less than 1ms either way.
This information was more of a trivia.

> What's a rule of thumb to know when useMemo is reducing performance vs not? 

Usually `useMemo` should be used when:
- there is an actual performance issue
- the computed value is an object (i.e. not a [primitive value](https://developer.mozilla.org/en-US/docs/Glossary/Primitive) - with the exception of [Symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol)) that is used as a dependency of another hook (if object reference changes on every re-render, the hooks that depend on it will have to be re-evaluated on every re-render, causing even more re-renders down the component tree).

Otherwise it's usually better to extract a function out of the component, i.e. here it'd be:
```tsx
const labelError = getLabelError(internalValue);
```

> How can I time this myself?

Usually there is no need to time this.
If the computation is trivial (like here) there is no need to optimize anything.
One should default to not using hooks if they're not necessary (this applies to all hooks).

To measure the actual performance you can use Profiler from [React Developer Tools extension](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi) and see how it affects the time spent rendering a given component.